lib/perl/Genome/Disk/Detail/Allocation/Mover.pm

```diff
@@ -94,19 +94,79 @@ sub move {
+                $shadow_allocation->delete;
+                confess(
+                    $allocation_object->error_message(
+                        sprintf(
+                            "Could not move original allocation path (%s) to final path (%s)."
+                                . "  Differences found between directories: %s",
+                            $shadow_allocation_abs_path,
+                            $new_volume_final_path,
+                            $diff,
+                        )
+                    )
+                );
+            }
+            $self->warning_message(
+                "removing existing path (%s), confirmed redundant with original path (%s)",
+                $new_volume_final_path,
+                $original_absolute_path
+            );
+            # No differences detected, remove new path to allow rename of shadow allocation
+            unless (Genome::Sys->remove_directory_tree($new_volume_final_path)) {
+                $allocation_lock->unlock();
+                my $shadow_allocation_abs_path = $shadow_allocation->absolute_path;
+                $shadow_allocation->delete;
+                confess(
+                    $allocation_object->error_message(
+                        sprintf(
+                            "Could not remove existing, redundant final path (%s).",
+                            $new_volume_final_path
+                        )
+                    )
+                );
+            }
+        } else {
+            # Something exists, but not a directory
+            $allocation_lock->unlock();
+            my $shadow_allocation_abs_path = $shadow_allocation->absolute_path;
+            $shadow_allocation->delete;
+            confess(
+                $allocation_object->error_message(
+                    sprintf(
+                        "Unexpected non-directory path (%s) already exists!",
+                        $new_volume_final_path
+                    )
+                )
+            );
+        }
+    }
 
     Genome::Sys->create_directory($new_volume_final_path);
     unless (Genome::Sys->rename($shadow_allocation->absolute_path, $new_volume_final_path)) {
         $allocation_lock->unlock();
         my $shadow_allocation_abs_path = $shadow_allocation->absolute_path;
         $shadow_allocation->delete;
-        confess($allocation_object->error_message(sprintf(
-                "Could not move shadow allocation path (%s) to final path (%s)."
-                . "  This should never happen, even when 100%% full.",
-                $shadow_allocation_abs_path,
-                $new_volume_final_path)));
+        confess(
+            $allocation_object->error_message(
+                sprintf(
+                    "Could not move shadow allocation path (%s) to final path (%s)."
+                        . "  This should never happen, even when 100%% full.",
+                    $shadow_allocation_abs_path,
```

Comment:
This one wasn't part of this PR, but it could also be changed to use the existing variable :smile: